### PR TITLE
Col data replacement

### PR DESCRIPTION
**Pull request by HelenaLC**
- added unit-tests for `colData<-`
- `colData<-` protects `sample_id` & `spatialDataNames`
  - replacement by NULL silently keeps these fields 
  - replacement of `sample_id`s required one-to-one mapping and is `stop()`ped otherwise
  - valid replacement updates the `sample_id` field in `imgData`
- `spatialData<-` protects `colData`
  - replacement by NULL drops only `spatialDataNames` fields
  - valid replacement updates the `spatialDataNames` field in `int_metadata`

### CODE REVIEW

File: R/SpatialExperiment-methods.R [hunk 1]
```diff
@@ -175,8 +175,10 @@ setReplaceMethod("spatialData",
 setReplaceMethod("spatialData", 
     c("SpatialExperiment", "NULL"),
     function(x, value) {
+        # @<- is required here to enforce replacement because
+        # spatialDataNames are protected during colData replacement
         i <- spatialDataNames(x)
-        colData(x)[i] <- NULL
```
Comment (drighelli) [inline R/SpatialExperiment-methods.R:179]: Just a question, so we decided to entirely drop the `colData` and the `spatialData`?
Sorry, I don't remember the entire discussion ...

**Review by drighelli — COMMENTED** [review]
I left a comment on the spatialExperiment-methods.R

File: R/SpatialExperiment-methods.R (hunk 1 above)
Comment (HelenaLC) [reply R/SpatialExperiment-methods.R:179]: Hm, I think we discussed to separate the two. So one can drop the `spatialData` (which are just other cell metadata (`colData`), e.g. `in_tissue` might be irrelevant at some point). However, `spatialCoords` are kept separate from this, similar to a `reducedDim`. I think that makes sense - but yes, it's a different way of thinking than what we had previously. However, given that most computations will eventually run on `spatialCoords` (plus other non-`spatialData` cell metadata), I think this is sensible.

Comment (drighelli) [reply R/SpatialExperiment-methods.R:179]: ok for the spatialCoords, it makes sense. 
But the instruction `x@colData[i] <- NULL` drops also the `spatialData` right?

Comment (HelenaLC) [reply R/SpatialExperiment-methods.R:179]: Yes, for the `spatialData()<-NULL` replacement methodm anything specified as `spatialDataNames` in `int_metadata()$spatialDataNames` will be dropped. Does that make sense or did you mean something else?

**Review by drighelli — APPROVED** [review]

**Pull request merged**